src/realm/object-store/sync/sync_session.cpp

```diff
@@ -742,6 +742,16 @@ void SyncSession::handle_error(sync::SessionErrorInfo error)
                 next_state = NextStateAfterError::inactive;
                 log_out_user = true;
                 break;
+            case sync::ProtocolErrorInfo::Action::MigrateSchema:
+                std::function<void()> callback;
+                {
+                    util::CheckedLockGuard l(m_state_mutex);
+                    callback = m_sync_schema_migration_callback;
```

Comment:
Should you "move" or clear the callback when you grab it to call it?

Reply:
sure, let's do that.